Core/Business/Product/Form/ProductInformation.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Business\Product\Form;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints as Assert;
+use PrestaShop\PrestaShop\Core\Foundation\Form\Type\TranslateType;
+use PrestaShop\PrestaShop\Core\Foundation\Form\Type\DropFilesType;
+use PrestaShop\PrestaShop\Core\Business\Form\Type\ChoiceCategoriesTreeType;
+use PrestaShop\PrestaShop\Core\Business\Form\Type\TypeaheadProductCollectionType;
+use PrestaShop\PrestaShop\Core\Foundation\IoC\Container;
+
+/**
+ * This form class is risponsible to generate the basic product informations form
+ */
+class ProductInformation extends AbstractType
+{
+    private $router;
+    private $context;
+    private $translator;
+    private $tax_rules;
+    private $manufacturers;
+    private $locales;
+    private $nested_categories;
+    private $productAdapter;
+
+    /**
+     * Constructor
+     *
+     * @param Container $container
+     */
+    public function __construct(Container $container)
+    {
+        $this->router = $container->make('Routing');
+        $this->context = $container->make('Context');
+
+        $this->translator = $container->make('Translator');
+        $this->productAdapter = $container->make('CoreAdapter:Product\\ProductDataProvider');
+        $this->locales = $container->make('CoreAdapter:Language\\LanguageDataProvider')->getLanguages();
+        $this->nested_categories = $container->make('CoreAdapter:Category\\CategoryDataProvider')->getNestedCategories();
```

Comment:
This is a very nice step in the right direction! While you're there, couldn't you pass the context to `getNestedCategories()` so that it doesn't have to guess the correct shop, language etc. from unsafe globals?